plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/unzipping/GeneralLayerUnzipper.java

```diff
@@ -0,0 +1,245 @@
+            LEdge nextEdgeToSplit = incomingEdge;
+            for (int layerIndex = 0; layerIndex < targetLayer; layerIndex++) {
+                LNode dummyNode = createDummyNode(graph, nextEdgeToSplit);
+                if (nodeIndex + edgeCount > subLayers.get(layerIndex).getNodes().size()) {
+                    dummyNode.setLayer(subLayers.get(layerIndex));
+                } else {
+                    dummyNode.setLayer(nodeIndex + edgeCount, subLayers.get(layerIndex)); 
+                }
+                nextEdgeToSplit = LongEdgeSplitter.splitEdge(nextEdgeToSplit, dummyNode);
+            }
+            if (targetLayer > 0) {
+                edgeCount += 1;
+            }
+        }
+        
+        // create unconnected dummy nodes to fill the layers if there are no incoming edges
+        if (noIncomingEdges) {
+            for (int layerIndex = 0; layerIndex < targetLayer; layerIndex++) {
+                LNode dummyNode = new LNode(graph);
+                dummyNode.setType(NodeType.PLACEHOLDER);
+                if (nodeIndex + edgeCount > subLayers.get(layerIndex).getNodes().size()) {
+                    dummyNode.setLayer(subLayers.get(layerIndex));
+                } else {
+                    dummyNode.setLayer(nodeIndex + edgeCount, subLayers.get(layerIndex));
+                }
+            }
+            if (targetLayer > 0) {
+                edgeCount += 1;
+            }
+        }
+        
+        // handle outgoing edges and following layers
+        boolean extraEdge = false;
+        for (LEdge outgoingEdge : node.getOutgoingEdges()) {
+            LEdge nextEdgeToSplit = outgoingEdge;
+            for (int layerIndex = targetLayer + 1; layerIndex < subLayers.size(); layerIndex++) {
+                LNode dummyNode = createDummyNode(graph, nextEdgeToSplit);
+                dummyNode.setLayer(subLayers.get(layerIndex));
+                nextEdgeToSplit = LongEdgeSplitter.splitEdge(nextEdgeToSplit, dummyNode);
+            }
+
+            for (int layerIndex = 0; layerIndex <= targetLayer; layerIndex++) {
+                if (extraEdge) {
+                    // add a placeholder beneath node's old position so that later
+                    LNode placeholder = new LNode(graph);
+                    placeholder.setType(NodeType.NONSHIFTING_PLACEHOLDER);
+                    
+                    if (nodeIndex + 1 > subLayers.get(layerIndex).getNodes().size()) {
+                        placeholder.setLayer(subLayers.get(layerIndex));
+                    } else {
+                        placeholder.setLayer(nodeIndex + 1, subLayers.get(layerIndex));
+                    }
+                }
+            }
+
+            if (extraEdge) {
+                edgeCount += 1;
+            }
+            
+            extraEdge = true;
+        }
+
+        if (edgeCount > 0) {
+            return edgeCount - 1;
+        } else {
+            return 0;
+        }
+    }
+
+    /**
```

Comment:
Missing javadoc.

Reply:
I've updated the javadoc and I think this is ready to merge then, I've added the PR to the 0.9.2 milestone.